ci/get-engine.ps1

```diff
@@ -0,0 +1,102 @@
+param(
+  [string] $gdk_home = (get-item "$($PSScriptRoot)").parent.FullName, ## The root of the UnrealGDK repo
+  [string] $gcs_publish_bucket = "io-internal-infra-unreal-artifacts-production/UnrealEngine"
+)
+
+$ErrorActionPreference = 'Stop'
+
+function Write-Log() {
+  param(
+    [string] $msg,
+    [Parameter(Mandatory=$false)] [bool] $expand = $false
+  )
+  if ($expand) {
+      Write-Output "+++ $($msg)"
+  } else {
+      Write-Output "--- $($msg)"
+  }
+}
+
+function Start-Event() {
+    param(
+        [string] $event_name,
+        [string] $event_parent
+    )
+
+    # Start this tracing span.
+    Start-Process -NoNewWindow "imp-ci" -ArgumentList @(`
+        "events", "new", `
+        "--name", "$($event_name)", `
+        "--child-of", "$($event_parent)"
+    ) | Out-Null
+
+    Write-Log "--- $($event_name)"
+}
+
+function Finish-Event() {
+    param(
+        [string] $event_name,
+        [string] $event_parent
+    )
+
+    # Emit the end marker for this tracing span.
+    Start-Process -NoNewWindow "imp-ci"  -ArgumentList @(`
+        "events", "new", `
+        "--name", "$($event_name)", `
+        "--child-of", "$($event_parent)"
+    ) | Out-Null
+}
+
+pushd "$($gdk_home)"
+
+    # Fetch the version of Unreal Engine we need
+    pushd "ci"
+        if (Test-Path env:ENGINE_COMMIT_HASH)
+        {
+            $unreal_version = Get-Item env:ENGINE_COMMIT_HASH
+            Write-Log "Using engine version defined by ENGINE_COMMIT_HASH: $($unreal_version)"
+        } else {
+            $unreal_version = Get-Content -Path "unreal-engine.version" -Raw
+            Write-Log "Using engine version found in unreal-engine.version file: $($unreal_version)"
+        }
+    popd
+
+    ## Create an UnrealEngine directory if it doesn't already exist
+    New-Item -Name "UnrealEngine" -ItemType Directory -Force
+
+    Start-Event "download-unreal-engine" "build-unreal-gdk-:windows:"
+    pushd "UnrealEngine"
+        Write-Log "Downloading the Unreal Engine artifacts from GCS"
```

Comment:
Perhaps move this log till after the gcs_unreal_location definiton and include it so we know what file it was looking for if it failed.